tutorials/haskell/hspec-golden-test-tutorial/tutorial.md

```diff
@@ -0,0 +1,506 @@
+
+So, now we know what golden testing is, but if unit tests and golden tests are so similar, why don't we just keep up with unit tests, right?
+
+
+The truth is that unit tests are not useful when evaluating complex and large outputs and that's where the difference lies. Test are all about maintainability and readability and when we tests large outputs we tend to add visual noise to our code. Helpfully Golden Tests fixes this issue, lets check the main advantages.
+
+ - Large outputs are stored in a golden file reducing the visual noise in our code.
+
+ - With some golden tests libraries the expected output (Golden File) can be automatically generated.
+
+ - Some golden test libraries also automatically updates the golden file. So, we don't need to manually update the tests when the expected output changes. 
+
+ - It is useful when working with JSON, HTML, images, etc.
+
+In this tutorial we will use [hspec-golden](http://hackage.haskell.org/package/hspec-golden) library to build golden tests. We will also use the ```hspec-golden``` CLI to easily update them.
+
+You can either code along with this tutorial or check at the finished [code](https://github.com/stackbuilders/tutorials/tree/tutorials/tutorials/haskell/hspec-golden-test-tutorial/code) in Github. 
+
+## Hspec-golden
+
+`Hspec-golden` is a StackBuilders testing library, written in haskell, that helps users implement golden tests. It is an OpenSource project so feel free to [review](https://github.com/stackbuilders/hspec-golden) it.
+
+## Getting Started
+
+First things first, lets create a new Haskell project on which we will be working:
+
+```bash
+stack new "hspec-golden-tests"
+```
+
+and add the necessary dependencies in order to follow this tutorial without issues.
+
+Project dependencies
+
+``` yaml
+dependencies:
+- base >= 4.7 && < 5
+- aeson
+- text
+- bytestring
+- blaze-html
+```
+
+Testing dependencies
+
+``` yaml
+dependencies:
+- hspec-golden-tests
+- hspec
+- hspec-golden
+
+```
+## A Simple Test
+
+Now that we have all dependencies installed, lets get started with golden tests.
+
+First lets create a ```FizzBuzz ``` module to be tested and add some code. For convinience I have created it inside a ```FIZZBUZZ``` directory. 
+
+```haskell
+module FIZZBUZZ.FizzBuzz where 
+
+fizzBuzz :: [Int] -> [String]
+fizzBuzz list = map fizzOrBuzz list
+
+fizzOrBuzz ::Int -> String
+fizzOrBuzz n | n `mod` 15 == 0  = "FizzBuzz"
+             | n `mod` 3  == 0  = "Fizz"
+             | n `mod` 5  == 0  = "Buzz"
+             | otherwise        = show n
+
+```
+For simplicity we will test a simple function `fizzbuzz` that replaces 3 multiples with fizz and 5 multiples with buzz, in case a number is multiple of both it replaces the number with fizzbuzz. But before moving on with testing, lets review some of ```hspec-golden``` [documentation](http://hackage.haskell.org/package/hspec-golden-0.1.0.1/docs/Test-Hspec-Golden.html).
```

Comment:
@Jagl257 Shouldn't it be _multiples of 3_ and _multiples of 5_? Also, Could you please capitalize `buzz`, `fizz` and `fizzbuzz`? (e.g. `Buzz`)

Reply:
Yes, indeed. Thank's @CristhianMotoche